pinot-core/src/main/java/org/apache/pinot/core/operator/query/SelectionOnlyOperator.java

```diff
@@ -132,7 +132,11 @@ protected SelectionResultsBlock getNextBlock() {
           Object[] values = blockValueFetcher.getRow(docId);
           for (int colId = 0; colId < numExpressions; colId++) {
             if (_nullBitmaps[colId] != null && _nullBitmaps[colId].contains(docId)) {
-              values[colId] = null;
+              if (values[colId] instanceof Double) {
+                values[colId] = Double.NEGATIVE_INFINITY;
+              } else {
+                values[colId] = null;
+              }
```

Comment:
I don't think this is correct. Do we need to change this?

Reply:
I think we want this behaviour for decimals only. For other flows, behaviour must be as it is currently.

Reply:
When `null` is enabled, we should be able to handle `null` value properly. Do you observe issues if reverting this?

Reply:
You are right. This change is not needed.